### PR TITLE
Mobs pushing eachother wont change their direction

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
This will be pretty much critical if we get the vision thing merged.

Otherwise it doesn't matter much either way, but I think it makes more sense people would stumble backwards rather than two people rapidly facing back and forth as they bump into eachother.

:cl: Kor
fix: You will now retain your facing when getting pushed by another mob.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -188,7 +188,12 @@
 					return
 		if(pulling == AM)
 			stop_pulling()
+		var/current_dir
+		if(isliving(AM))
+			current_dir = AM.dir
 		step(AM, t)
+		if(current_dir)
+			AM.dir = current_dir
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: No, ```setDir()```

**Comment by MrPerson** [comment]
One thing I would love to see in the future is a Move() argument for a facing direction so that the mob would be facing in that direction while calling all the movement subprocs.

This is def a good change though, 100%.

**Comment by Jordie0608** [comment]
You should make a changelog for this.

**Comment by PKPenguin321** [comment]
I actually think it might be cool to try vision cones without this, intentionally shove your enemy so you can jump them from behind while they're briefly turned away

**Comment by KorPhaeron** [comment]
@PKPenguin321 

It'd be shitty and annoying in actual combat rapidly being shoved back and forth though

**Comment by PKPenguin321** [comment]
Yeah but it requires running into melee range from a cardinal direction which is usually a no no in combat

I think it's worth trying at least? I could be dead wrong but you never know

**Comment by Supermichael777** [comment]
strafing when. actually putting strafing and sprint in as a part of a movement overhaul isn't such a bad idea if this gets merged. just got to add nausea for spinning around too much

**Comment by anconfuzedrock** [comment]
strafing more like fixed eye amirite guys

**Pull request merged**